core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -45,17 +48,17 @@ public DataExpressionSource(final ValueExpression dataExpression, final int inde
     protected byte[] getData(final long offset, final int size) throws IOException {
         final ImmutableList<Optional<Value>> results = dataExpression.eval(graph, encoding);
         if (results.size <= index) { throw new IllegalStateException("ValueExpression dataExpression yields " + results.size + " result(s) (expected at least " + (index + 1) + ")."); }
-        final byte[] inputData = getValueAtIndex(results, index, 0).get().getValue();
+        final byte[] inputData = getValueAtIndex(results, index, 0).computeResult().get().getValue();
```

Comment:
Yes, it is a bit tied to the style mentioned above. The method itself does not check it's parameters. But again, although I'd personally do it, for private methods I think it can be left as is.